dbms/src/Storages/DeltaMerge/ColumnFile/ColumnFileSchema.h

```diff
@@ -0,0 +1,146 @@
+// Copyright 2022 PingCAP, Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <Common/nocopyable.h>
+#include <Core/Block.h>
+#include <Interpreters/Context.h>
+#include <Storages/BackgroundProcessingPool.h>
+#include <Storages/DeltaMerge/DeltaMergeDefines.h>
+#include <common/types.h>
+#include <openssl/base.h>
+#include <openssl/sha.h>
+
+#include "boost/container_hash/hash_fwd.hpp"
+
+namespace std
+{
+using Digest = UInt256;
+template <>
+struct hash<Digest>
+{
+    size_t operator()(const Digest & digest) const
+    {
+        size_t seed = 0;
+        boost::hash_combine(seed, boost::hash_value(digest.a));
+        boost::hash_combine(seed, boost::hash_value(digest.b));
+        boost::hash_combine(seed, boost::hash_value(digest.c));
+        boost::hash_combine(seed, boost::hash_value(digest.d));
+        return seed;
+    }
+};
+} // namespace std
+
+namespace DB
+{
+namespace DM
+{
+using Digest = UInt256;
+class ColumnFileSchema
+{
+private:
+    Block schema;
+
+    using ColIdToOffset = std::unordered_map<ColId, size_t>;
+    ColIdToOffset colid_to_offset;
+
+public:
+    explicit ColumnFileSchema(const Block & schema_)
+        : schema(schema_)
+    {
+        for (size_t i = 0; i < schema.columns(); ++i)
+            colid_to_offset.emplace(schema.getByPosition(i).column_id, i);
+    }
+
+    const DataTypePtr & getDataType(ColId column_id) const
+    {
+        // Note that column_id must exist
```

Comment:
Please put this note as a doc comment of the function, so that caller can know it without looking at the impl, for example:

```
/// Returns the data type of a column. The specified column id must exist, otherwise ....
```